storybookjs/storybook
# Core: Fix `useStoryPrepared` hook failing with `undefined` data

Closes #




## What I did



`api.getData` returns undefined sometimes. which is used by isPrepared, the inner logic of useStoryPrepared hook.
however, isPrepared hook access to 'type' att without any check.
so I added an exception for this. Now hooks works as expected.


## How to test



1.  Clone this branch.
2.  Type this where can use manager-api`import { useStoryPrepared } from '@storybook/manager-api` 
3.  Check hooks works without error

 it 

## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`





### 🦋 Canary release


This pull request has been released at the npm dist tag `pr-22631`. Install it by pinning all your Storybook dependencies to that tag.
<details>
<summary>More information</summary>

| | |
| --- | --- |
| **Published version** | [`7.1.1-pr-22631-1689802540-351503cb.0`](https://npmjs.com/package/@storybook/cli/v/7.1.1-pr-22631-1689802540-351503cb.0) |
| **npm dist tag** | `@storybook/cli@pr-22631` |
| **Triggered by** | @JReinhold |
| **Repository** | [SpookyJelly/storybook](https://github.com/SpookyJelly/storybook) |
| **Branch** | [`patch-1`](https://github.com/SpookyJelly/storybook/tree/patch-1) |
| **Commit** | [`351503cb`](https://github.com/SpookyJelly/storybook/commit/351503cb9a6718a79bdbf0a42f1bdb0515f8bd44) |
| **Datetime** | Wed Jul 19 21:35:40 UTC 2023 (`1689802540`) |
| **Workflow run** | [5604333170](https://github.com/storybookjs/storybook/actions/runs/5604333170) |

To request a new release of this PR ping the `@storybookjs/core` team.

_core team members can create a new canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=22631`_
</details>
